src/js/grapl-cdk/lib/schemadb.ts

```diff
@@ -12,63 +11,66 @@ export interface SchemaDbProps {
+    readonly schema_table: DbTable;
+    readonly schema_properties_table: DbTable;
 
     constructor(scope: cdk.Construct, id: string, props: SchemaDbProps) {
         super(scope, id);
 
-        this.schema_table = new dynamodb.Table(this, 'SchemaDbTable', {
+        this.schema_table = new DbTable(this, 'SchemaDbTable', {
             tableName: props.edges_table_name,
+            table: 'schema_db_table',
             partitionKey: {
                 name: 'f_edge',
                 type: dynamodb.AttributeType.STRING,
             },
-            serverSideEncryption: true,
-            billingMode: dynamodb.BillingMode.PAY_PER_REQUEST,
-            removalPolicy: RemovalPolicy.DESTROY,
+            tableReference: 'SchemaDbTable',
         });
 
-        this.schema_properties_table = new dynamodb.Table(this, 'SchemaPropertiesDbTable', {
-            tableName: props.properties_table_name,
-            partitionKey: {
-                name: 'node_type',
-                type: dynamodb.AttributeType.STRING,
-            },
-            serverSideEncryption: true,
-            billingMode: dynamodb.BillingMode.PAY_PER_REQUEST,
-            removalPolicy: RemovalPolicy.DESTROY,
-        });
+        this.schema_properties_table = new DbTable(
+            this,
+            'SchemaPropertiesDbTable',
+            {
+                tableName: props.properties_table_name,
+                table: 'schema_properties_db_table',
+                partitionKey: {
+                    name: 'node_type',
+                    type: dynamodb.AttributeType.STRING,
+                },
+                tableReference: 'SchemaPropertiesDbTable',
+            }
+        );
     }
 
     allowRead(service: Service|FargateService) {
         for (const table of [this.schema_table, this.schema_properties_table]) {
             if (service instanceof FargateService) {
-                table.grantReadData(service.service.taskDefinition.taskRole);
-                table.grantReadData(service.retryService.taskDefinition.taskRole);
+                table.table.grantReadData(service.service.taskDefinition.taskRole);
+                table.table.grantReadData(service.retryService.taskDefinition.taskRole);
             } else {
-                table.grantReadData(service.event_handler);
-                table.grantReadData(service.event_retry_handler);
+                table.table.grantReadData(service.event_handler);
+                table.table.grantReadData(service.event_retry_handler);
             }
         }
     }
 
+
     allowReadWrite(service: Service) {
         for (const table of [this.schema_table, this.schema_properties_table]) {
-            table.grantReadWriteData(service.event_handler);
-            table.grantReadWriteData(service.event_retry_handler);
+            table.table.grantReadWriteData(service.event_handler);
+            table.table.grantReadWriteData(service.event_retry_handler);
```

Comment:
All of these `table.table`s can jsut be `table.` - since your DynamoDBTable type also exposes `grant*Data` functions!

Reply:
I thought so, too. I tried removing the extra table to be sure and get a message "Property 'grantReadData' does not exist on type 'DbTable'.ts(2339)" - I think we poorly named the for loop variable, but I also can't think of something more creative.